### PR TITLE
Stapler 1.250

### PR DESCRIPTION
**Pull request by jglick**
[Changes](https://github.com/stapler/stapler/compare/stapler-parent-1.248...stapler-parent-1.250)

@reviewbybees

### CODE REVIEW

**Comment by daniel-beck** [comment]
Could we maybe get a more thorough review on this one? IIRC both previous Stapler upgrades broke Jenkins.

**Comment by daniel-beck** [comment]
Actual regression.

**Comment by jglick** [comment]
@i386 I am guessing this is a consequence of https://github.com/stapler/stapler/pull/96; do you plan to take a look?

**Comment by jglick** [comment]
Or perhaps from https://github.com/stapler/stapler/commit/bd67c1eda4d096c0943a37ff1e84f028bf6348f5, which does not seem to have been done via PR at all.

**Comment by jglick** [comment]
Confirmed with `git-bisect` that https://github.com/stapler/stapler/pull/96 is responsible. Working on a fix.

**Comment by i386** [comment]
Nice catch @jglick

File: changelog.html
Comment (egok) [inline changelog.html]: why 1.249 instead of 1.250?

Comment (jglick) [inline changelog.html]: Well, 1.249 had the fix for the stated bug, 1.250 just corrected an obscure regression introduced by an unrelated change in 1.249.

**Comment by daniel-beck** [comment]
Given how problematic recent Stapler upgrades were, this shouldn't be merged into the weekly release that will serve as the base for next week's security update.

**Review by svanoort — APPROVED** [review]
🐝  -- I've looked at the changelog and nothing is jumping out at me as immediately problematic (though that may not mean much, since I'm not familiar with common Stapler problem areas).  

Built it, ran it locally, installed plugins and created+ran jobs -- seems to run fine, plugins install, etc. 

Not sure what else to do to properly exercise this one?

**Comment by jglick** [comment]
@reviewbybees done

> Not sure what else to do to properly exercise this one?

Great, that is more than most PRs will get.

**Comment by oleg-nenashev** [comment]
@svanoort @jglick I propose to ensure there is no binary compatibility breakages before merging. Ideally we also need to ensure that ATH/PCT do not become more broken with this change.

**Comment by jglick** [comment]
> ensure there is no binary compatibility breakages before merging

How do you propose to do that?

> ensure that ATH/PCT do not become more broken

Without merging? How?

File: changelog.html
Comment (oleg-nenashev) [inline changelog.html]: I would vote for 1.250. And it is better to just remove changelog change from the PR

**Comment by oleg-nenashev** [comment]
1) Code review. I went through changes, and they seem to be fine from the compatibility perspective at least
2) Just build a custom core locally and run tests against it.

**Comment by jglick** [comment]
Sorry but I do not have the time or resources to run a potentially unbounded set of tests against a local build. Since when is that a prerequisite for merging changes to core?

**Comment by oleg-nenashev** [comment]
It is not a prerequisite, but we had pretty bad regressions on this year due to Stapler upgrades. 

Feel free to merge if you are ready to take responsibility for handling the fallout if something goes wrong. I did it once on this year, and the Stapler maintainers (@kohsuke and @vivek) did not follow the commitments they made after that fallout. @jglick and we would have another regression, which has been prevented by you.

So I am not comfortable about merging before we test it using the tools we have.

**Comment by jglick** [comment]
> if you are ready to take responsibility for handling the fallout if something goes wrong

Of course if any regression is discovered I would do my best to respond at once.

**Comment by daniel-beck** [comment]
If this breaks things again (for the fifth time in a year!) I'm not going to be happy.

**Pull request merged**
